src/dune_engine/dep.ml

```diff
@@ -77,15 +77,17 @@ module Fact = struct
     type t =
       { files : Digest.t Path.Map.t
       ; dirs : Digest.t Path.Map.t (* Only for file selectors for now *)
-      ; parent_dirs : Path.Set.t
+      ; (* XXX these directories are only needed for sandboxing, perhaps they
+           should be generated lazily? *)
+        dirs_that_must_exist : Path.Build.Set.t
```

Comment:
It doesn't seem right that `files` and `dirs_that_must_exist`, which are derived from `files` have different types. Why not use `dirs_that_must_exist : Path.t` here?

I think I actually prefer keeping `parent_dirs : Path.Set.t` in this type because that seems to have clearer meaning.

The `dirs_that_must_exist` name seems only useful for the function exported in the `mli`.

Reply:
I guess, you may be trying to cache as much work as possible in this field. But is it really worth the trouble if we only use this for sandboxing? The cost of running an action in the sandbox will surely outweigh anything we might possible win here. So, maybe we should just drop this field altogether.

Reply:
> It doesn't seem right that files and dirs_that_must_exist, which are derived from files have different types. Why not use dirs_that_must_exist : Path.t here?

Well because for sandboxing we only care about the build paths. The external paths are assumed to be created.

> I think I actually prefer keeping parent_dirs : Path.Set.t in this type because that seems to have clearer meaning.

Sure, but it is less accurate.

> I guess, you may be trying to cache as much work as possible in this field. But is it really worth the trouble if we only use this for sandboxing? The cost of running an action in the sandbox will surely outweigh anything we might possible win here. So, maybe we should just drop this field altogether.

Okay, I can do that.

How about changing the name to reflect sandboxing. For example, `val necessary_dirs_for_sandboxing : t -> Path.Build.Set.t`. That seems better to me than accumulating unnecessary paths.

Reply:
Thanks! Agreed with the last idea.

Reply:
Done.